versioned_docs/version-stable/nym-simple-websocket-tutorial.md

```diff
@@ -0,0 +1,658 @@
+
+    <img src="/img/tutorials/simple-websocket/image1.png"/>
+
+    You will then notice a `package.json` file has been created in the directory:
+
+    <img src="/img/tutorials/simple-websocket/image2.png"/>
+
+ 2. Continuing with our terminal, type and enter:
+    
+    ```
+    npm install typescript
+    ```
+    After this point , we should open up our chosen IDE (VSCode, Sublime Text, etc) and open up the folder we are working in (the one with the `package.json`).
+
+    <img src="/img/tutorials/simple-websocket/image3.png"/>
+
+    We can see that typescript has been added to our `dependencies`. Typescript is now in our project.
+
+
+ 3. Back in our terminal, type and enter:
+    
+    ```
+    npm install ts-node --save-dev
+    ```
+    This package (`ts-node`) allows us to build a typescript application in a node environment.
+
+ 4. Create a new file in the same level as `package.json` called `tsconfig.json` Inside that file , copy and paste the code below into it.
+
+    ```
+    {
+        "compilerOptions": {
+            "module": "commonjs",
+            "esModuleInterop": true,
+            "target": "es6",
+            "moduleResolution": "node",
+            "sourceMap": true,
+            "outDir": "dist"
+        },
+        "lib": ["es2015"]
+    }
+    ```
+ 5. Now that we have got to this point, we want to be able to run our application on localhost to make sure everything is working okay. 
+    We also want to be able to work on our application while its running and make sure it automatically picks up any saved changes when we view it on the browser. To do this , we are going to use the Parcel bundler.
+
+    Back in your terminal, type and enter:
+
+    ```
+    npm install --global parcel-bundler
+    ```
+
+    After the npm install has finished, create a new folder called src in the same level as our tsconfig.json and package.json (our project root). Inside there , create 2 new files. One called `index.html` and one called `index.ts`.
+
+ 6. In our `index.html`, paste in the following code:
+    
+    ```
+    <!DOCTYPE html>
+    <html>
+        <head>
+            <title>App Test</title>
+            <meta charset="utf-8"/>
+        </head>
+        <body>
+            <h1>Test</h1>
+            <div id="app"></div>
+            <script src="index.ts"></script>
+        </body>
+    </html>
+
+    ```
+
+    And in our index.ts, paste the following code:
+
+    ```
+    console.log('test log')
+    ```
+
+    Once we have done that , navigate into our `package.json` and in the scripts section, just above the test command,  paste the following:
+
+    ```
+    "start": "parcel src/index.html"
+    ```
+
+    Now back in our terminal , type `npm start` and press enter.
+
+    <img src="/img/tutorials/simple-websocket/image4.png"/>
+
+    This will result in the above output. Open your browser to see the results (localhost:1234). You should now have a running web application!
+
+    
+### Building the User Client
```

Comment:
Needs a snappier / more descriptive name: the client-side code? a 'user client' is a bit vague. This makes more sense in combination with the comment above also